### PR TITLE
Activate all-access APKs

### PR DESCRIPTION
**Pull request by nirvn**
@3nids , as discussed, you'll get test APKs from this.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 15f6397473063473d7a82049dbed213164353ed8: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-x86-android.apk)
</details>

**Comment by nirvn** [comment]
@3nids , arm 64 all access link: http://qfieldapks.sos-ch-dk-2.exo.io/ci-builds/qfield-plus-dev-3519-2b6c60f8ab5bcbb90bc81b0932c5ff21bca1aac3-arm64-android.apk

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -25,6 +25,14 @@ jobs:
...
           - triplet: 'x86-android'
             all_files_access: 'OFF'
+          - triplet: 'arm64-android'
+            all_files_access: 'ON'
+          - triplet: 'arm-android'
+            all_files_access: 'ON'
+          - triplet: 'x64-android'
+            all_files_access: 'ON'
+          - triplet: 'x86-android'
+            all_files_access: 'ON'
```
Comment (3nids) [inline .github/workflows/android.yml:36]: you can simplify a bit the matrix:
```suggestion
    strategy:
      matrix:
        triplet: ['arm64-android', 'arm-android', 'x64-android', 'x86-android']
        all_files_access: ['ON', 'OFF']
```

**Comment by nirvn** [comment]
@3nids , alright, updated the workflow to add the all access APKs in the auto-generated comment. Let's see if it works.

I've also changed the suffix from "plus" to "all-access". I felt "plus" was giving a completely wrong sense of that version being somehow _better or coming with extra functionality_ when it fact it's merely about storage access.

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -155,11 +148,11 @@ jobs:
           path: |
             ${{ env.CMAKE_BUILD_DIR }}/android-build/build/outputs/apk/release/android-build-release-signed.apk
 
-      - name: 📦 Upload plus artifacts
+      - name: 📦 Upload all access artifacts
         uses: actions/upload-artifact@v3
         if: ${{ matrix.all_files_access }} == "ON"
         with:
-          name: android-plus-${{ matrix.triplet }}
+          name: android-all-access-${{ matrix.triplet }}
```
Comment (3nids) [inline .github/workflows/android.yml:155]:
```suggestion
          name: ${{ ENV.APP_PACKAGE_NAME }}-android-${{ matrix.triplet }}
```

you can use this, and remove the other upload artifact step.

File: scripts/ci/upload_artifacts.sh [hunk 3]
```diff
@@ -10,16 +10,16 @@ then
   if [[ "${CI_TAG}" ]];
   then
     if [[ ${ALL_FILES_ACCESS} == "ON" ]]; then
-      #FILENAME_AAB="qfield-${CI_TAG}-${TRIPLET}.aab"
-      FILENAME_APK="qfield-${CI_TAG}-${TRIPLET}.apk"
+      #FILENAME_AAB="qfield-all-access-${CI_TAG}-${TRIPLET}.aab"
+      FILENAME_APK="qfield-all-access-${CI_TAG}-${TRIPLET}.apk"
```
Comment (3nids) [inline scripts/ci/upload_artifacts.sh:14]: this part is already handled in the other script, no?
you should be able to remove the conditions and just use `APP_PACKAGE_NAME`

Comment (3nids) [reply scripts/ci/upload_artifacts.sh:14]: basically:

```
  if [[ "${CI_TAG}" ]];
  then
    #FILENAME_AAB="${APP_PACKAGE_NAME}-${CI_TAG}-${TRIPLET}.aab"
    FILENAME_APK="${APP_PACKAGE_NAME}-${CI_TAG}-${TRIPLET}.apk"
  else
    #FILENAME_AAB="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.aab"
    FILENAME_APK="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
  fi
```

File: .github/workflows/android.yml [hunk 4]
```diff
@@ -144,15 +144,7 @@ jobs:
         uses: actions/upload-artifact@v3
         if: ${{ matrix.all_files_access }} == "OFF"
```
Comment (3nids) [inline .github/workflows/android.yml:145]: You'll need to remove this one

File: .github/workflows/android.yml [hunk 5]
```diff
@@ -243,6 +228,10 @@ jobs:
             - [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_PACKAGE_FILE_BASENAME }}-arm-android.apk)
             - [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_PACKAGE_FILE_BASENAME }}-x64-android.apk)
             - [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_PACKAGE_FILE_BASENAME }}-x86-android.apk)
+            - [all access arm64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/${{ env.CI_ALL_ACCESS_PACKAGE_FILE_BASENAME }}-arm64-android.apk)
```
[the remark this answers is not in the document]
Comment (3nids) [reply .github/workflows/android.yml:230]: seems fixed now

File: scripts/ci/env_gh.sh [hunk 6]
```diff
@@ -44,22 +44,24 @@ fi
...
+  export APP_PACKAGE_NAME="qfield_all_access${APP_PACKAGE_NAME_SUFFIX}"
+  export CI_PACKAGE_FILENAME="qfield-all-access-${CI_PACKAGE_FILE_SUFFIX}-${ARCH}.apk"
 else
-  export CI_PACKAGE_FILE_BASENAME="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}"
-  if [[ ${ALL_FILES_ACCESS} == "ON" ]]; then
-    export APP_PACKAGE_NAME="qfield_plus_dev"
-  else
-    export APP_PACKAGE_NAME="qfield_dev"
-  fi
+  export APP_PACKAGE_NAME="qfield_all_access${APP_PACKAGE_NAME_SUFFIX}"
```
Comment (3nids) [inline scripts/ci/env_gh.sh:59]:
```suggestion
  export APP_PACKAGE_NAME="qfield${APP_PACKAGE_NAME_SUFFIX}"
```

File: scripts/ci/upload_artifacts.sh [hunk 7]
```diff
@@ -9,21 +9,11 @@ then
...
   else
-    if [[ ${ALL_FILES_ACCESS} == "ON" ]]; then
-      #FILENAME_AAB="qfield-plus-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.aab"
-      FILENAME_APK="qfield-plus-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
-    else
-      #FILENAME_AAB="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.aab"
-      FILENAME_APK="qfield-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
-    fi
+    #FILENAME_AAB="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.aab"
+    FILENAME_APK="${APP_PACKAGE_NAME}-dev-${CI_UPLOAD_ARTIFACT_ID}-${CI_COMMIT}-${TRIPLET}.apk"
```
Comment (3nids) [inline scripts/ci/upload_artifacts.sh:16]: actually you can remove line 10 to 17 by this
```suggestion
    FILENAME_APK="${APP_PACKAGE_NAME}-${CI_PACKAGE_FILE_SUFFIX}-${TRIPLET}.apk"
```

Comment (3nids) [reply scripts/ci/upload_artifacts.sh:16]: it's all defined in env_gh

Comment (nirvn) [reply scripts/ci/upload_artifacts.sh:16]: @3nids , yeah, true.

**Review by 3nids — APPROVED** [review]

**Pull request merged**